core/src/main/kotlin/xtdb/cache/Stats.kt

```diff
@@ -1,3 +1,22 @@
 package xtdb.cache
 
-data class Stats(val pinnedBytes: Long, val evictableBytes: Long, val freeBytes: Long)
+interface Stats {
+    val pinnedBytes: Long
+    val evictableBytes: Long
+    val freeBytes: Long
+
+    companion object {
+
+        fun new (pinnedBytes: Long, evictableBytes: Long, freeBytes: Long): Stats {
```

Comment:
I'd probably avoid the name `new` here - I know it's not a keyword in Kotlin but still probably a name I'd avoid on the JVM :sweat_smile: 

one idea - I've used `operator fun invoke(...)` here in the past, so that you can call it as `Stats(...)`.